pkg/controller/install/webhook.go

```diff
@@ -69,80 +69,105 @@ func (i *StrategyDeploymentInstaller) createOrUpdateWebhook(caPEM []byte, desc v
 }
 
 func (i *StrategyDeploymentInstaller) createOrUpdateMutatingWebhook(ogNamespacelabelSelector *metav1.LabelSelector, caPEM []byte, desc v1alpha1.WebhookDescription) error {
-	webhooks := []admissionregistrationv1.MutatingWebhook{
-		desc.GetMutatingWebhook(i.owner.GetNamespace(), ogNamespacelabelSelector, caPEM),
+	webhookSelector := labels.SelectorFromSet(ownerutil.OwnerLabel(i.owner, i.owner.GetObjectKind().GroupVersionKind().Kind)).String()
+	existingWebhooks, err := i.strategyClient.GetOpClient().KubernetesInterface().AdmissionregistrationV1().MutatingWebhookConfigurations().List(context.TODO(), metav1.ListOptions{LabelSelector: webhookSelector})
+	if err != nil {
+		return err
+	}
+
+	var webhook *admissionregistrationv1.MutatingWebhookConfiguration
+	for _, item := range existingWebhooks.Items {
+		if strings.HasPrefix(item.Name, desc.Name) {
```

Comment:
What happens if one prefix is a prefix of another, e.g. `prefix1` and `prefix12`? Won't `prefix12` match both descriptions?

Reply:
This has been resolved in the latest commit by:

- Adding label to Webhooks that are based on the webhook description name
- Making sure that all Webhooks have unique names.